sites/kit.svelte.dev/src/lib/search/SearchBox.svelte

```diff
@@ -0,0 +1,416 @@
+		$searching = false;
+	}
+
+	$: if ($searching) update();
+
+	$: recent_searches = lookup ? $recent.map((href) => lookup.get(href)).filter(Boolean) : [];
+</script>
+
+<svelte:window
+	on:keydown={(e) => {
+		if (e.code === 'KeyK' && (navigator.platform === 'MacIntel' ? e.metaKey : e.ctrlKey)) {
+			e.preventDefault();
+			$query = '';
+			$searching = !$searching;
+		}
+
+		if (e.code === 'Escape') {
+			$searching = false;
+		}
+	}}
+/>
+
+{#if $searching && index}
+	<div
+		class="modal-background"
+		on:click={() => ($searching = false)}
+		on:wheel={(e) => e.preventDefault()}
+		on:touchmove={(e) => e.preventDefault()}
+	/>
+
+	<div
+		bind:this={modal}
+		class="modal"
+		on:keydown={(e) => {
+			if (e.key === 'ArrowDown' || e.key === 'ArrowUp') {
+				e.preventDefault();
+				const group = focusable_children(e.currentTarget);
+
+				// when using arrow keys (as opposed to tab), don't focus buttons
+				const selector = 'a, input';
+
+				if (e.key === 'ArrowDown') {
+					group.next(selector);
+				} else {
+					group.prev(selector);
+				}
+			}
+
+			if (e.code === 'Space' && e.target.nodeName !== 'INPUT') {
+				e.preventDefault();
+			}
+		}}
+		use:trap
+	>
+		<div class="search-box">
+			<!-- svelte-ignore a11y-autofocus -->
+			<input
+				autofocus
+				on:keydown={(e) => {
+					if (e.key === 'Enter') {
+						if (results.length > 0) {
+							modal.querySelector('a').click();
+						}
+					}
+
+					// ideally, opening the modal would create a history entry that we
+					// could use with browser back/forward buttons — this would be the
+					// best way to allow people to close the modal on mobile, for
+					// example. That's a little tricky to do right now — need to add
+					// new APIs to SvelteKit — but in the meantime we can close the
+					// modal when the backspace key is pressed. we _don't_ want that
+					// to happen when the backspace keydown fires repeatedly, so we
+					// track whether it was already pressed or not. (Sadly this
+					// doesn't work on mobile, longpress will close the modal)
+					if (e.key === 'Backspace') {
+						if ($query === '' && !backspace_pressed) {
+							$searching = false;
+						}
+
+						backspace_pressed = true;
+					}
+				}}
+				on:keyup={(e) => {
+					if (e.key === 'Backspace') {
+						backspace_pressed = false;
+					}
+				}}
+				on:input={(e) => {
+					$query = e.target.value;
+					update();
+				}}
+				value={$query}
+				placeholder="Search"
+			/>
+
+			<button aria-label="Close" on:click={() => ($searching = false)}>
+				<Icon name="close" />
+			</button>
+
+			<ul class="results">
+				{#if $query}
+					{#each results as result, i}
+						<!-- svelte-ignore a11y-mouse-events-have-key-events -->
+						<li>
+							<a on:click={() => navigate(result.href)} href={result.href}>
+								<small>{result.breadcrumbs.join('/')}</small>
+								<strong>{@html excerpt(result.title, $query)}</strong>
+								<span>{@html excerpt(result.content, $query)}</span>
```

Comment:
Fixed the first by using `--second` everywhere instead of `--flash`:

<img width="622" alt="image" src="https://user-images.githubusercontent.com/1162160/152848644-979f84de-dc88-4836-a375-9ac02fc666fb.png">

The second is a bit trickier — could use `--text` everywhere, but it would make the results look very noisy, which would make them much _harder_ to parse for people for whom the current contrast is sufficient. Not sure what the best solution is, will keep fiddling